### PR TITLE
Add prefix setting for posh-git default prompt.

### PR DESCRIPTION
**Pull request by rkeithhill**
Add Pesters tests for the prompt function.  Update docs for the new prefix and the fact that abbreviate home dir is disabled by default.

### CODE REVIEW

File: src/en-US/about_posh-git.help.txt [hunk 1]
```diff
@@ -34,14 +39,28 @@ GIT TAB COMPLETION
...
 
-GIT STATUS PROMPT
-    PowerShell generates its prompt by executing a prompt function, if one
-    exists. posh-git defines such a function in the included
-    profile.example.ps1 script file that outputs the current working directory
-    followed by an abbreviated git status summary e.g.:
+POWERSHELL PROMPT
+    PowerShell generates its prompt by executing a function named "prompt", if
+    one exists. posh-git will install its prompt function if it detects the
+    user does not have their own, customized prompt fucntion. This prompt
```
Comment (dahlbyk) [inline src/en-US/about_posh-git.help.txt]: s/fucntion/function/

File: src/en-US/about_posh-git.help.txt [hunk 2]
```diff
@@ -34,14 +39,28 @@ GIT TAB COMPLETION
...
+    displays the current working directory followed by git status summary
+    information if the current directory is located in a Git repository e.g.:
```
Comment (dahlbyk) [inline src/en-US/about_posh-git.help.txt]: comma before "e.g."?

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 posh-git is a PowerShell module that integrates Git and PowerShell by providing Git status summary information that can be displayed in the PowerShell prompt e.g.:
```
Comment (dahlbyk) [inline README.md]: Comma before "e.g."?

File: src/en-US/about_posh-git.help.txt [hunk 4]
```diff
@@ -105,10 +131,24 @@ GIT STATUS PROMPT
...
+    prompt function. Prompt formatting, among other things, can be customized
+    using the global variables: $GitPromptSettings, $GitTabSettings and
+    $TortoiseGitSettings. To see the available settings, simply type the
+    variable name at the PowerShell prompt (after posh-git has been imported)
+    and press Enter.
+
+    For more information on customizing the posh-git default prompt or creating
+    your own prompt fuction see:
+    https://github.com/dahlbyk/posh-git/wiki/Customizing-Your-PowerShell-Prompt
+
```
Comment (dahlbyk) [inline src/en-US/about_posh-git.help.txt]: Extra newline?

File: test/DefaultPrompt.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            $res | Should Match "$escapedHome \d+ms> "
+        }
+    }
+
+    Context 'Prompt with Git summary' {
+        BeforeAll {
+            Set-Location $PSScriptRoot
+
+            function global:git {
+                $cmdline = "$args"
```
Comment (dahlbyk) [inline test/DefaultPrompt.Tests.ps1]: Does `$OFS = ''` set above mess with this?

Comment (dahlbyk) [inline test/DefaultPrompt.Tests.ps1]: More generally, is this mock `git` function something that should be defined in Shared?

**Comment by rkeithhill** [comment]
> who's going to write a Markdown-to-PowerShell-help converter

It would be nice with the ANSI support if PowerShell could render basic markdown in the console.  But a converter would be nice too.  Take a look at the change.  I cleaned up the typos and tweaked the tests some more.  There were failing when I ran them from PowerShell where I already had a prompt/settings/posh-git loaded.  These changes make this scenario work although it leaves with the default posh-git prompt.

**Pull request merged**